### ISSUE TITLE
qgis-28 build stopped working suddenly

### ISSUE THREAD

**Issue by jkopera** [OSGeo/homebrew-osgeo4mac#90]
Hi All,

qgis 2.8.02 was working beautiful until today, when, for some reason, libproj.0.dylib disappeared.  Reinstalled both qgis and proj, still bringing up same error.  Thoughts?  

Would hate to have to uninstall all my homebrew os4geo stuff in order to use Kyngchaos's version and not have conflicts up the wazoo.

Echo on upgrading the formula to 2.8.1  I am not ruby savvy, unfortunately, so have no clue how to go about doing it myself.

Process:               QGIS [3491]
Path:                  /usr/local/Cellar/qgis-28/2.8.0_2/QGIS.app/Contents/MacOS/QGIS
Identifier:            org.qgis.qgis2-hb
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           QGIS [3491]
User ID:               501

Date/Time:             2015-04-22 15:11:16.031 -0400
OS Version:            Mac OS X 10.10.3 (14D136)
Report Version:        11
Anonymous UUID:        EA57DBDC-9739-44E3-54CA-CBF3A8050300

Time Awake Since Boot: 1200 seconds

Crashed Thread:        0

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000002, 0x0000000000000000

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /usr/local/lib/libproj.0.dylib
  Referenced from: /usr/local/Cellar/qgis-28/2.8.0_2/QGIS.app/Contents/MacOS/QGIS
  Reason: image not found

Binary Images:
    0x7fff649c7000 -     0x7fff649fd837  dyld (353.2.1) <65DCCB06-339C-3E25-9702-600A28291D0E> /usr/lib/dyld
    0x7fff8b2b4000 -     0x7fff8b326fff  com.apple.framework.IOKit (2.0.2 - 1050.20.2) <09C0518C-90DF-3FC3-96D6-34D35F72C8EF> /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
    0x7fff90910000 -     0x7fff90ca8ff7  com.apple.CoreFoundation (6.9 - 1153.18) <5C0892B8-9691-341F-9279-CA3A74D59AA0> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff92815000 -     0x7fff92815fff  com.apple.ApplicationServices (48 - 48) <5BF7910B-C328-3BF8-BA4F-CE52B574CE01> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices

Model: iMac13,2, BootROM IM131.010A.B04, 4 processors, Intel Core i7, 3.4 GHz, 32 GB, SMC 2.11f14
Graphics: NVIDIA GeForce GTX 680MX, NVIDIA GeForce GTX 680MX, PCIe, 2048 MB
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1600 MHz, 0x0707, -
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1600 MHz, 0x0707, -
Memory Module: BANK 0/DIMM1, 8 GB, DDR3, 1600 MHz, 0x0707, -
Memory Module: BANK 1/DIMM1, 8 GB, DDR3, 1600 MHz, 0x0707, -
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0xF4), Broadcom BCM43xx 1.0 (7.15.166.24.3)
Bluetooth: Version 4.3.4f4 15601, 3 services, 27 devices, 1 incoming serial ports
Network Service: Ethernet, Ethernet, en0
Serial ATA Device: APPLE HDD ST3000DM001, 3 TB
Serial ATA Device: APPLE SSD SM128E, 121.33 GB
USB Device: AS2105
USB Device: Hub
USB Device: USB2.0 Hub
USB Device: Microsoft® Nano Transceiver v2.0
USB Device: iPhone
USB Device: FaceTime HD Camera (Built-in)
USB Device: Hub
USB Device: Hub
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus: iMac, Apple Inc., 23.4

**Comment by pjduplooy** [comment]
jkopera

This happened to me too.  What I suspect is that with the newest brew update, proj (libproj) got updated to 4.9.1 (was 4.8.0).  libproj.0.dylib is now libproj.9.dylib.  I just made a link (ln -sf libproj.0.dylib libproj.9.dylib) and everything worked perfectly.  This is not perfect but anyway.

Concerning 2.8.1 I did the following:

Change /usr/local/Library/Formula/qgis-28.rb so that the lines containing url and sha1 now reads:

  url "https://github.com/qgis/QGIS/archive/final-2_8_1.tar.gz"
  sha1 "78a8eed91a5e151da82db9142c2ee5b131b974d6"

Incidentally my qgis was installed as follows:

brew -v install qgis-28 --with-orfeo, --with-grass7, --with-r, --with-saga-gis, --without-server.

Hope this helps.

Pieter

**Comment by jkopera** [comment]
Hi Pieter,

Thank you very much for the insight and help!  I'll see if this works today.  

Thanks again,
Joe

Sent via mobile phone

_______________________________
Joseph P. Kopera
Geologist and Senior Research Fellow
Massachusetts Geological Survey
University of Massachusetts, Amherst
jkopera@geo.umass.edu
http://www.geo.umass.edu/stategeologist
_______________________________

**Comment by rbanick** [comment]
Hi All,

I had the same problem after upgrading this morning. Running @pjduplooy 's fix unfortunately did nothing for me. The same error message appeared after linking within the usr/local/lib folder

Going the more manual route of dropping in a duplicate **libproj.9.dylib** file, renamed as **libproj.0.dylib**, unfortunately just created the following error message:

````
  Library not loaded: /usr/local/lib/libproj.0.dylib
  Referenced from: /usr/local/Cellar/qgis-28/2.8.0_2/QGIS.app/Contents/MacOS/QGIS
  Reason: no suitable image found.  Did find:
	/usr/local/lib/libproj.0.dylib: stat() failed with errno=62
	/usr/local/lib/libproj.0.dylib: stat() failed with errno=62
````

I reversed this change after seeing the error message as it seemed to be only digging me deeper. I'm not sure why the link isn't working. I dug into the cellar and tried to fix the link there, to no avail. Any thoughts?

Best,
Robert

**Comment by rbanick** [comment]
Slight update:

After running ```ln -sf /usr/local/lib/libproj.9.dylib /usr/local/lib/libproj.0.dylib```

I got the error message from before:

```
  Library not loaded: /usr/local/lib/libproj.0.dylib
  Referenced from: /usr/local/Cellar/qgis-28/2.8.0_2/QGIS.app/Contents/MacOS/QGIS
  Reason: no suitable image found.  Did find:
    /usr/local/lib/libproj.0.dylib: stat() failed with errno=62
    /usr/local/lib/libproj.0.dylib: stat() failed with errno=62
````

Still stumped what to do from here though. I am not the most fluent in the deep intricacies of dependencies.

**Comment by rbanick** [comment]
Got this to work. Beforehand, I uninstalled proj, resintalled as proj-4.8.0 (via the .rb file), watched that break all the other dependencies, uninstalled proj-4.8.0 and reinstalled the newest version of proj. After all the experiments, rerunning ```ln -sf /usr/local/lib/libproj.9.dylib /usr/local/lib/libproj.0.dylib``` in the usr/local/lib folder resolved this after all. 

No clue why this didn't work the first time around but I'm not complaining.

**Comment by jmking** [comment]
I'm also having this problem. Linking did not work. I'm not quite sure of all the steps rbanick detailed above so have not yet tried those

**Comment by jmking** [comment]
Update: Did the same thing as rbanick (I think anyway), ran brew uninstall proj then brew install proj. After that the linking solution seems to work. Anyone know why that would be?

**Comment by julionovoa** [comment]
Linking ```ln -sf /usr/local/lib/libproj.9.dylib /usr/local/lib/libproj.0.dylib``` worked. Thanks!

**Comment by EconGeo** [comment]
I also had the same problem using the bottled version.  I did ```brew rm qgis-28```, then built from source with ```brew install qgis-28 --with-grass7 --with-orfeo``` after a failed build attempt with ```--with-globe``` and the build found ```proj 4.9.1``` so it appears the formula is building fine but the bottled version needs to be updated or perhaps require an upgrade of ```proj``` when installing an updated bottle.

**Comment by jarandaf** [comment]
@rbanick @julionovoa ++

**Comment by smnorris** [comment]
Thanks, linking works for me too

**Comment by micheleorsi** [comment]
Linking that files worked also for me! Thanks

**Comment by Ylannl** [comment]
The symlink works for me as well.

**Comment by dakcarto** [comment]
Thanks! Fixed in d0d87f5. Bottles to be added shortly.
